lib/library.rb

```diff
@@ -0,0 +1,27 @@
+class Library
+
+  attr_reader :name, :books, :authors, :start, :end
+
+  def initialize(library_name)
+    @name = library_name
+    @books = []
+    @authors = []
+  end
+
+  def add_author(incoming_author)
+    @authors << incoming_author.name
+    @books << incoming_author.books
+  end
+
+  def publication_time_frame_for(given_author)
+    date_range = []
+    given_author.books.each do |book|
+      date_range << book.publication_year
+    end
+    date_range.sort
+    start_end_range = []
+    :start == date_range[0]
+    :end == date.range.last
+    start_end_range = :start, :end
```

Comment:
I'm not totally sure where you were going with lines 22-25, however up through line 21 looks good! If we add a variable to line 21 so we can use that result, we can finish up this method like this:
```suggestion
    sorted_range = date_range.sort
    { start: sorted_range.first,
      end: sorted_range.last }
```